src/modules/sync/generateSyncModel.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  AssetContracts,
+  ContentItemContracts,
+  ContentTypeContracts,
+  ContentTypeSnippetContracts,
+  ManagementClient,
+  TaxonomyContracts,
+} from "@kontent-ai/management-sdk";
+
+import { ManagementClientBaseOptions } from "../../types/managementClient.js";
+import { transformContentTypeModel } from "./modelTransfomers/contentTypes.js";
+import { transformContentTypeSnippetsModel } from "./modelTransfomers/contentTypeSnippets.js";
+import { transformTaxonomyGroupsModel } from "./modelTransfomers/taxonomyGroups.js";
+import { FileContentModel } from "./types/fileContentModel.js";
+import { getRequiredIds } from "./utils/contentTypeHelpers.js";
+import {
+  fetchContentTypes,
+  fetchContentTypeSnippets,
+  fetchRequiredAssets,
+  fetchRequiredContentItems,
+  fetchTaxonomies,
+} from "./utils/fetchers.js";
+
+export type EnvironmentModel = {
+  taxonomyGroups: ReadonlyArray<TaxonomyContracts.ITaxonomyContract>;
+  contentTypeSnippets: ReadonlyArray<ContentTypeSnippetContracts.IContentTypeSnippetContract>;
+  contentTypes: ReadonlyArray<ContentTypeContracts.IContentTypeContract>;
+  assets: ReadonlyArray<AssetContracts.IAssetModelContract>;
+  items: ReadonlyArray<ContentItemContracts.IContentItemModelContract>;
+};
+
+export const fetchModel = async (config: ManagementClientBaseOptions): Promise<EnvironmentModel> => {
+  const client = new ManagementClient({
+    environmentId: config.environmentId,
+    apiKey: config.apiKey,
+  });
+
+  const contentTypes = await fetchContentTypes(client);
+  const contentTypeSnippets = await fetchContentTypeSnippets(client);
+  const taxonomies = await fetchTaxonomies(client);
+
+  const allIds = [...contentTypes, ...contentTypeSnippets].reduce<{ assetIds: Set<string>; itemIds: Set<string> }>(
+    (previous, type) => {
+      const ids = getRequiredIds(type);
+
+      return {
+        assetIds: new Set([...previous.assetIds, ...ids.assetIds]),
+        itemIds: new Set([...previous.itemIds, ...ids.itemIds]),
+      };
+    },
+    { assetIds: new Set<string>(), itemIds: new Set<string>() },
```

Comment:
Don't specify the generic type in the `Set` constructor. It should not be necessary.